README.md

```diff
@@ -73,6 +73,8 @@ module.exports = {
       // note: these settings are mutually exclusive and allowedFilesRegex has priority over skippedFilesRegex
       allowedFilesRegex: null, // RegExp to only target specific fonts by their names
       skippedFilesRegex: null, // RegExp to skip specific fonts by their names
+      textRegex: /\.(js|css|html)$/,  // RegExp for searching text reference
+      webpackCompilationHook: 'thisCompilation', // Webpack compilation hook (for example PurgeCss webpack plugin use 'compilation' )
```

Comment:
awesome! would you be willing to add a test and/or complete example to the docs of working with vue+purgecss?

Reply:
Sure I'll add a commit for updating the README.md. 
but shortly in this is my vue.config.js for vue3 with vue3 / tailwindcss 3 / fontawesome 6 / webpack 5 / purgecss-webpack-plugin / fontmin-webpack / mangle-css-class-webpack: 
```js
const webpackPlugins = [];

// **********************************
// Purgecss unused classes
//
if (process.env.__DEBUG__ !== '1') {
  const PurgecssPlugin = require('purgecss-webpack-plugin');
  const glob = require('glob-all')
  const PATHS = {
    src: path.join(__dirname, 'src')
  }
  const purgeCssPlugin = new PurgecssPlugin({
    paths: glob.sync(
      [
        path.join(__dirname, './public/*.html'),
        path.join(__dirname, './src/**/*.vue'),
        path.join(__dirname, './src/**/*.js')
      ]),
    safelist: [/^sm:/, /^md:/, /^lg:/, /^xl:/, /^2xl:/, /^focus:/, /^hover:/, /^group-hover:/, /\[.*\]/, /^basicLightbox/, /\/[0-9]/, /^tns/],
    fontFace: true
  })
  webpackPlugins.push(purgeCssPlugin);
}

// **********************************
// fontminifying Fontawesome
//
if (process.env.__DEBUG__ !== '1') {
  const FontMinPlugin = require('fontmin-webpack');
  const fontMinPlugin = new FontMinPlugin({
    autodetect: true,
    glyphs: [],
    allowedFilesRegex: /^fa[srltdb]*-/, // RegExp to only target specific fonts by their names
    skippedFilesRegex: null, // RegExp to skip specific fonts by their names
    textRegex: /\.(js|css|html|vue)$/,  // RegExp for searching text reference
    webpackCompilationHook: 'compilation', // Webpack compilation hook (for example PurgeCss webpack plugin use 'compilation' )
  });
  webpackPlugins.push(fontMinPlugin);
}

// **********************************
// Mangling class names
//
//this use my fork of mangle-css-webpack-plugin 
//https://github.com/eltorio/mangle-css-class-webpack-plugin.git
//take care of the classNameRegExp because you may accidentally mangle parts of the javascript code !
if (process.env.__DEBUG__ !== '1') {
  const MangleCssClassPlugin = require('mangle-css-class-webpack-plugin'); 
  const myManglePlugin = new MangleCssClassPlugin({
    classNameRegExp: '(bg|[-]*p[xylrbt]*|[-]*m[xylrbt]*|[-]*left|[-]*top|[-]*right|[-]*bottom|w|[-]*z|h|justify|overflow|border|max|flex|text|font|inline|rounded|from|to|via|contrast|brightness|leading|items|backdrop|shadow|duration|whitespace|self|cursor|transition|outline)-[a-z0-9_-]+|relative|static|absolute|shadow|flex|hidden|rounded|border',
    log: true,
    reserveClassName: ['fa', 'fas', 'far', 'p', 'm', 'z', 'pt', 'pb', 'px', 'py', 'pl', 'pr', 'mt', 'mb', 'mx', 'my', 'ml', 'mr', 'to'],
    ignorePrefixRegExp: '.*tns.*|light[bB]ox'
  });
  webpackPlugins.push(myManglePlugin);
}

module.exports = {
   runtimeCompiler: true,
   configureWebpack: {
    plugins: webpackPlugins,
    devtool: false,
    mode: 'production',
  },
};
```
in package.json 
```json
"devDependencies": {
…
    "fontmin-webpack": "file:fontmin-webpack",
    "glob-all": "^3.3.0",
    "mangle-css-class-webpack-plugin": "file:mangle-css-class-webpack-plugin",
    "purgecss-webpack-plugin": "^4.1.3",
    "webpack": "~5.71.0",
…
}